src/array_api_stubs/_draft/array_object.py

```diff
@@ -1089,11 +1098,13 @@ def __setitem__(
         ----------
         self: array
             array instance.
-        key: Union[int, slice, ellipsis, Tuple[Union[int, slice, ellipsis], ...], array]
+        key: Union[SupportsIndex, slice, ellipsis, Tuple[Union[SupportsIndex, slice, ellipsis], ...], array]
             index key.
         value: Union[int, float, bool, array]
             value(s) to set. Must be compatible with ``self[key]`` (see :ref:`broadcasting`).
 
+        .. note::
+           ``key`` can only be an array if it is valid for boolean array indexing, or supports ``__index__()``.
```

Comment:
I really can't figure out why this gives a warning—any playing around with notes for `__setitem__()` gives me the same thing  :thinking:

```
/home/honno/PycharmProjects/array-api/src/array_api_stubs/_draft/array_object.py:docstring of array_api_stubs._draft.array_object._array.__setitem__:1: WARNING: py:class reference target not found: ..
```

Reply:
There is `nitpick_ignore` list in `src/_array_api_conf.py` that may be needed here.

Reply:
Can ignore the re-review-request still need to figure this one out :sweat_smile: